daemon/api_model.go

```diff
@@ -69,3 +89,90 @@ func postModel(c *Command, r *http.Request, _ *auth.UserState) Response {
 	return AsyncResponse(nil, &Meta{Change: chg.ID()})
 
 }
+
+// helper for parsing url options into formatting option vars
+func parseHeadersFormatOptionsFromURL(url *url.URL) (
+	jsonResult bool,
+	headersOnly bool,
+	headers map[string]string,
+	err error,
+) {
+	headers = map[string]string{}
+	q := url.Query()
+	for k := range q {
+		if k == "json" {
+			switch q.Get(k) {
+			case "false":
+				jsonResult = false
+			case "headers":
+				headersOnly = true
+				fallthrough
+			case "true":
+				jsonResult = true
+			default:
+				err = errors.New(`"json" query parameter when used must be set to "true" or "headers"`)
+				return jsonResult, headersOnly, headers, err
+			}
+			continue
+		}
+		headers[k] = q.Get(k)
+	}
+
+	return jsonResult, headersOnly, headers, err
+}
+
+// getModel gets the current model assertion using the DeviceManager
+func getModel(c *Command, r *http.Request, _ *auth.UserState) Response {
+	jsonResult, headersOnly, _, err := parseHeadersFormatOptionsFromURL(r.URL)
+	if err != nil {
+		return BadRequest(err.Error())
+	}
+
+	devmgr := c.d.overlord.DeviceManager()
+
+	model, err := devmgr.Model()
```

Comment:
without locking state, this will panic every time.

You need to test this a lot more.

Reply:
Yes, in 2.39 where I was testing this, it was locking inside Model, but that is not the case on master anymore.